providers/aws/connection/awsutil.go

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package connection
+
+import (
+	"context"
+	"errors"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"golang.org/x/oauth2"
+	"google.golang.org/api/idtoken"
+)
+
+type TokenSourceProvider func(ctx context.Context) (oauth2.TokenSource, error)
+
+func AwsWithIdentityProvider(opts map[string]string) (*aws.Config, error) {
```

Comment:
this func is called `AwsWithidentityProvider` but internally it calls the GCpInstanceIdentity. Can we rename it to make it clear this only works with GCP?